reconcile/utils/ocm/service_log.py

```diff
@@ -12,16 +12,22 @@
 from reconcile.utils.ocm.search_filters import Filter
 from reconcile.utils.ocm_base_client import OCMBaseClient
 
+CLUSTER_SERVICE_LOGS_LIST_ENDPOINT = "/api/service_logs/v1/clusters/cluster_logs"
+CLUSTER_SERVICE_LOGS_CREATE_ENDPOINT = "/api/service_logs/v1/cluster_logs"
 
-def get_service_logs(
-    ocm_api: OCMBaseClient, filter: Filter
+
+def get_service_logs_for_cluster_uuid(
+    ocm_api: OCMBaseClient, cluster_uuid: str, filter: Optional[Filter] = None
```

Comment:
Is there a use case that `filter` is `None`?

Another case is what if an empty filter (`Filter()`) passed in, should `params["search"]` be populated as empty string or that key shouldn't be in the payload at all?

Reply:
this basically just represents the functionality of the endpoint which supports unfiltered requests since the `search` parameter is optional (therefore i'm not adding the param if there is no filter). there will be other usecases for this endpoint in upcoming work and we will see then how the interface of this function can evolve. i think it is ok to not have a filter but we could add some protection against too big responses.

empty filters are not supported. they raise an error on rendering. i added a test do show the behaviour.